### PR TITLE
Add spent time to referenced issue in commit message

### LINKED ISSUE #123: Milestone card view
Currently we can list the issues of one milestone, but I think if we can set some progress and provide a card view, we will manage it well. A milestone like a project on github. So if you create a milestone, you have created a project, it's no need an extra UI to do that.

### LINKED ISSUE #124: Commits and pull files
Before we can start a review, we can implement comment on one line of PRs at first.
I think this includes two sub issues. 

* [ ] ~Commit comment #3748~ see #4898
* [x] Pull files comment #3748

### PR DESCRIPTION
**Pull request by lafriks**
Add support to specify spent time in commit message that will be added to references issue/PR.

Allows to specify time only for issues/PR that will be closed.

Time spent must be specified after issue/PR number in message.

Time should start with `@` symbol and must contain time format, ex. `@1h10m` - would mean 1 hour and 10 minutes.

It is possible to log time for multiple issues, ex:

```
Fixed #123 spent @1h, refs #102, fixes #124 @1.5h
```

Following time units are supported:
* `mo` - month (equals to 4 weeks)
* `w` - week (equals to 5 days)
* `d` - day (equals to 8 hours)
* `h` - hour
* `m` - minutes

Screenshot:
image

### CODE REVIEW

**Comment by silverwind** [comment]
Seems a good feature but hard to discover. Maybe add some docs somewhere for all the triggers we support in commit title/message?

Once thing I've noticed it that one cannot commit time to an issue without closing it. I think that should be allowed:

image

**Comment by zeripath** [comment]
Agree with @silverwind - this will need some documentation. I guess it could go in to the FAQ?

**Comment by lafriks** [comment]
@silverwind yes it is meant to work for scenario when issue is fixed by commit. Later it could be improved to work on referenced issues also. I will try to work on some docs

**Comment by lafriks** [comment]
@silverwind documentation added in docs section `Automatically Linked References`

File: modules/repofiles/action.go [hunk 1]
```diff
@@ -19,6 +22,16 @@ import (
 	"code.gitea.io/gitea/modules/setting"
 )
 
+const (
+	secondsByMinute = float64(time.Minute / time.Second) // seconds in a minute
```
Comment (6543) [inline modules/repofiles/action.go:26]: isnt souch const's in *time* package?

Comment (6543) [reply modules/repofiles/action.go:26]: -> just asking didn't had a look jet ...

Comment (lafriks) [reply modules/repofiles/action.go:26]: Not ones I'm aware of 😉

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by lafriks** [comment]
:rocket:

**Pull request merged**